interfaces/mount/backend.go

```diff
@@ -79,7 +80,16 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 		return fmt.Errorf("cannot synchronize mount configuration files for snap %q: %s", snapName, err)
 	}
 	if err := UpdateSnapNamespace(snapName); err != nil {
-		return fmt.Errorf("cannot update mount namespace of snap %q: %s", snapName, err)
+		// try to discard the mount namespace but only if there aren't enduring daemons in the snap
+		for _, app := range snapInfo.Apps {
+			if app.Daemon != "" && app.RefreshMode == "endure" {
+				return fmt.Errorf("cannot update mount namespace of snap %q when trying to update it, and cannot discard it because it contains an enduring daemon: %s", snapName, err)
```

Comment:
```suggestion
				return fmt.Errorf("cannot update mount namespace of snap %q, and cannot discard it because it contains an enduring daemon: %s", snapName, err)
```